src/Controller/DefaultController.php

```diff
@@ -206,380 +204,343 @@ public function moderate_reviews()
-
-            $db->insert('arborcat_patron_pickup_request')
-            ->fields([
-              'uid' => $user->id(),
-              'pickupDay' => $pickupDay,
-              'timeSlot' => $timeSlot,
-              'pickupLocation' => $pickupLocation,
-              'contactEmail' => $contactEmail,
-              'contactPhone' => $contactPhone,
-              'contactSMS' => $contactSMS,
-              'timestamp' => time()
-            ])
-            ->execute();
-            $result['success'] = "Successfully added pickup request";
-        } else {
-            $result['error'] = 'You must be logged in to make a pickup request.';
-        }
-
-        return new JsonResponse($result);
-    }
+  // this needed? insert can be done on form submit
+  private function addPickupRequest($pickupLocation, $pickupDay, $timeSlot, $contactEmail, $contactPhone, $contactSMS) {
+    $user = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
+    if ($user->isAuthenticated()) {
+        $db = \Drupal::database();
 
-    public function hook_form_FORM_ID_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id)
-    {
-        // Modification for the form with the given form ID goes here. For example, if
-        // FORM_ID is "user_register_form" this code would run only on the user
-        // registration form.
-
-        // Add a checkbox to registration form about agreeing to terms of use.
-        $form['terms_of_use'] = array(
-            '#type' => 'checkbox',
-            '#title' => t("I agree with the website's terms and conditions."),
-            '#required' => true,
-        );
+        $db->insert('arborcat_patron_pickup_request')
+        ->fields([
+          'uid' => $user->id(),
+          'pickupDay' => $pickupDay,
+          'timeSlot' => $timeSlot,
+          'pickupLocation' => $pickupLocation,
+          'contactEmail' => $contactEmail,
+          'contactPhone' => $contactPhone,
+          'contactSMS' => $contactSMS,
+          'timestamp' => time()
+        ])
+        ->execute();
+        $result['success'] = "Successfully added pickup request";
+    } else {
+        $result['error'] = 'You must be logged in to make a pickup request.';
     }
 
-    /*
-    * Debugging routine to log to the <root folder>/LWKLWK.log
-    */
-    public function dblog(...$thingsToLog)
-    {
-        $lineToLog = '';
-        foreach ($thingsToLog as $item) {
-            $lineToLog = $lineToLog . ' ' . print_r($item, true);
-        }
-        // prepend date/time onto log line
-        $nowDateTime = new DrupalDateTime();
-        $dateTimeString = (string) $nowDateTime->format('Y-m-d H:i:s');
-        $completeLine = '[' . $dateTimeString . '] ' . $lineToLog . "\n";
-        error_log($completeLine, 3, "LWKLWK.log");
-    }
+    return new JsonResponse($result);
+  }
+
+  public function hook_form_FORM_ID_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
```

Comment:
This looks like sample form alter code that isn't needed

Reply:
Removed this code plus a testing method addPickupRequest, plus calls to the method.
JS now handles clicks on checkboxes of items in the form and clicks on the select_all checkbox. Common method now checks the number of checked items and displays the warning banner if above the specified max number of items - currently 6.